### PR TITLE
Custom class for bibliography style

### PR DESCRIPTION
**Pull request by eebasso**
## Summary

This PR creates a new class in `conf.py` that allows a custom bibliography style for the documentation generated by Sphinx.

## Additional Information

The custom style is configured thus far to have family names appear first followed by given names abbreviated. As an example, `Jean-Luc Vay` becomes `Vay, J.-L.`.

The `allbibs.bib` file is modified to enforce the hyphen in the name "Jean-Luc".

The effects of this PR can be seen in the bibliography generated in `theory/intro`.
Before: 
Screenshot 2023-12-06 033512

After: 
Screenshot 2023-12-06 033616

### CODE REVIEW

File: Docs/source/conf.py [hunk 1]
```diff
@@ -58,7 +61,18 @@
 
 # Relative path to bibliography file, bibliography style
 bibtex_bibfiles = ['./refs.bib']
-bibtex_default_style = 'unsrt'
+
+class WarpXBibStyle(UnsrtStyle):
```
Comment (ax3l) [inline Docs/source/conf.py:72]: Cool! Can you please add inline comments or class doc strings to the reference for these options?

File: Docs/source/conf.py [hunk 2]
```diff
@@ -30,6 +30,8 @@
 import sys
 import urllib.request
 
+import pybtex.plugin
+from pybtex.style.formatting.unsrt import Style as UnsrtStyle
```
Comment (ax3l) [inline Docs/source/conf.py:34]: Please add the required extra package(s) to both the pip and spack requirements file in `Docs/`.

File: Docs/source/conf.py (hunk 1 above)
Comment (eebasso) [reply Docs/source/conf.py:72]: Yes, no problem. I'll add some comments and a link to the Sphinx documentation.

File: Docs/source/conf.py (hunk 2 above)
Comment (eebasso) [reply Docs/source/conf.py:34]: Should be done now.

**Review by ax3l — APPROVED** [review]
Thank you! :)

Btw, can we also have `J-L Vay, A Huebl, R Lehe, ...` as formatting? I find the double comma way in `Vay, J.-L., Huebl, A., ...` aesthetically a bit unfortunate. `Vay J-L, Huebl A, ...` would also be ok for me.

Can be a follow-up.

**Comment by eebasso** [comment]
Yes that's no problem. Just need to comment out the 'lastfirst' option. PR incoming shortly.

**Pull request merged**